src/shared/components/app/navbar.tsx

```diff
@@ -443,67 +429,16 @@ export class Navbar extends Component<NavbarProps, NavbarState> {
     return amAdmin() || moderatesS;
   }
 
-  fetchUnreads() {
-    poll(async () => {
-      if (window.document.visibilityState !== "hidden") {
-        const auth = myAuth();
-        if (auth) {
-          this.setState({
-            unreadInboxCountRes: await HttpService.client.getUnreadCount({
-              auth,
-            }),
-          });
-
-          if (this.moderatesSomething) {
-            this.setState({
-              unreadReportCountRes: await HttpService.client.getReportCount({
-                auth,
-              }),
-            });
-          }
-
-          if (amAdmin()) {
-            this.setState({
-              unreadApplicationCountRes:
-                await HttpService.client.getUnreadRegistrationApplicationCount({
-                  auth,
-                }),
-            });
-          }
-        }
-      }
-    }, updateUnreadCountsInterval);
-  }
-
-  get unreadInboxCount(): number {
-    if (this.state.unreadInboxCountRes.state == "success") {
-      const data = this.state.unreadInboxCountRes.data;
-      return data.replies + data.mentions + data.private_messages;
-    } else {
-      return 0;
-    }
+  get unreadInboxCount() {
+    return InboxService.unreadInboxCount;
   }
 
-  get unreadReportCount(): number {
-    if (this.state.unreadReportCountRes.state == "success") {
-      const data = this.state.unreadReportCountRes.data;
-      return (
-        data.post_reports +
-        data.comment_reports +
-        (data.private_message_reports ?? 0)
-      );
-    } else {
-      return 0;
-    }
+  get unreadReportCount() {
+    return InboxService.unreadReportCount;
   }
 
-  get unreadApplicationCount(): number {
-    if (this.state.unreadApplicationCountRes.state == "success") {
-      const data = this.state.unreadApplicationCountRes.data;
-      return data.registration_applications;
-    } else {
-      return 0;
-    }
+  get unreadApplicationCount() {
```

Comment:
You can remove these three `get` functions, and just reference the `InboxService`